src/main/java/com/google/devtools/build/lib/exec/TestPolicy.java

```diff
@@ -99,6 +96,10 @@ public Map<String, String> computeTestEnvironment(
     // Setup any test-specific env variables; note that this does not overwrite existing values for
     // TEST_RANDOM_SEED or TEST_SIZE if they're already set.
     testAction.setupEnvVariables(env, timeout);
+
+    // Rule-specified test env.
+    testAction.getExtraTestEnv().resolve(env, clientEnv);
```

Comment:
I would recommend moving this above the statement right above it, otherwise you could override vars such as `TEST_TIMEOUT`.

Reply:
done